src/plugins/console/server/routes/api/console/autocomplete_entities/index.ts

```diff
@@ -6,4 +6,194 @@
+        method: 'GET',
+        headers: { ...headers },
+        host: sanitizeHostname(hostname),
+        port: port === '' ? undefined : parseInt(port, 10),
+        protocol,
+        path: `${path}?pretty=false`, // add pretty=false to compress the response by removing whitespace
+      };
+
+      try {
+        const req = client.request(options, (res) => {
+          const chunks: Buffer[] = [];
+          res.on('data', (chunk) => {
+            chunks.push(chunk);
+
+            // Destroy the request if the response is too large
+            if (Buffer.byteLength(Buffer.concat(chunks)) > MAX_RESPONSE_SIZE) {
+              req.destroy();
+              reject(Boom.badRequest(`Response size is too large for ${path}`));
+            }
+          });
+          res.on('end', () => {
+            const body = Buffer.concat(chunks).toString('utf8');
+            resolve(JSON.parse(body));
+          });
+        });
+        req.on('error', reject);
+        req.end();
+        break;
+      } catch (err) {
+        if (idx === hosts.length - 1) {
+          reject(err);
+        }
+        // Try the next host
+      }
+    }
+  });
+};
+
+export const registerAutocompleteEntitiesRoute = (deps: RouteDependencies) => {
+  deps.router.get(
+    {
+      path: '/api/console/autocomplete_entities',
+      options: {
+        tags: ['access:console'],
+      },
+      validate: false,
+    },
+    async (context, request, response) => {
+      const settings = parse(request.url.search, { parseBooleans: true }) as unknown as Settings;
+
+      // If no settings are provided return 400
+      if (Object.keys(settings).length === 0) {
+        return response.badRequest({
+          body: 'Request must contain a query param of autocomplete settings',
+        });
+      }
+
+      const legacyConfig = await deps.proxy.readLegacyESConfig();
+      const configWithHeaders = {
+        ...legacyConfig,
+        headers: request.headers,
+        kibanaVersion: deps.kibanaVersion,
+      };
+
+      // Wait for all requests to complete, in case one of them fails return the successfull ones
+      const results = await Promise.allSettled([
+        getMappings(settings, configWithHeaders),
+        getAliases(settings, configWithHeaders),
+        getDataStreams(settings, configWithHeaders),
+        getLegacyTemplates(settings, configWithHeaders),
+        getIndexTemplates(settings, configWithHeaders),
+        getComponentTemplates(settings, configWithHeaders),
+      ]);
+
+      const [mappings, aliases, dataStreams, legacyTemplates, indexTemplates, componentTemplates] =
+        results.map((result) => {
+          // If the request was successful, return the result
+          if (result.status === 'fulfilled') {
+            return result.value;
+          }
+
+          // If the request failed, log the error and return an empty object
+          if (result.reason instanceof Error) {
+            deps.log.error(result.reason);
```

Comment:
Maybe we should change this to `debug` level? WDYT? I'm not sure if I'd consider it a critical error, and it could flood the logs if the autocomplete interval is running every minute.